target/linux/mediatek/image/filogic.mk

```diff
@@ -138,3 +138,20 @@ define Device/xiaomi_redmi-router-ax6000
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
 endef
 TARGET_DEVICES += xiaomi_redmi-router-ax6000
+
+define Device/xiaomi_redmi-router-ax6000-stock
+  DEVICE_VENDOR := Xiaomi
+  DEVICE_MODEL := Redmi Router AX6000 (stock layout)
+  DEVICE_DTS := mt7986a-xiaomi-redmi-router-ax6000-stock
+  DEVICE_DTS_DIR := ../dts
+  KERNEL_LOADADDR := 0x48000000
+  UBINIZE_OPTS := -E 5
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+ifneq ($(CONFIG_TARGET_ROOTFS_INITRAMFS),)
+  ARTIFACTS := initramfs-factory.ubi
+  ARTIFACT/initramfs-factory.ubi := append-image initramfs-kernel.bin | ubinize-kernel
```

Comment:
see here https://github.com/openwrt/openwrt/blob/master/target/linux/mediatek/image/mt7622.mk#L89
and here https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/image/generic.mk#L1125

Reply:
Ah sorry, I misread, the ifneq guarding `CONFIG_TARGET_ROOTFS_INITRAMFS` makes sense, though I believe it is always enabled on this target anyway.
To not break the ImageBuilder you still have to use `append-image-stage`.

Reply:
> To not break the ImageBuilder you still have to use `append-image-stage` (instead of `append-image`).

This is because the ImageBuilder doesn't build the initramfs image (no matter if CONFIG_TARGET_ROOTFS_INITRAMFS is set or not), but it will try to build all artifacts for the board. So this will fail and break ImageBuilder unless you use `append-image-stage`.